mne/encoding/model.py

```
@@ -0,0 +1,238 @@
+        all values per row must be `mask_val` for removal. Defaults to 'all'.
+    """
+    def __init__(self, estimator, mask_val=None, samples_train=None,
+                 samples_pred=None, mask_condition='all'):
+        if all(ii is not None for ii in [mask_val, samples_train]):
+            raise ValueError('Supply one of mask_val | samples_train, or both')
+        if samples_train is not None:
+            samples_train = np.asarray(samples_train).astype(int)
+        self.samples_train = samples_train
+        if samples_pred is not None:
+            samples_pred = np.asarray(samples_pred).astype(int)
+        self.samples_pred = samples_pred
+
+        self.mask_val = mask_val
+        if mask_condition not in ['all', 'any']:
+            raise ValueError('condition must be one of "all" or "any"')
+        self.mask_condition = mask_condition
+        self.est = _check_estimator(estimator)
+
+    def fit(self, X, y):
+        """Remove datapoints then fit the estimator.
+
+        Parameters
+        ----------
+        X : array, shape (n_samples, n_features)
+            The input array for the model. Samples (rows) will be
+            removed prior to fit according to `mask_val` and `samples_train`.
+        y : array, shape (n_samples, n_targets)
+            The output array for the model. Samples (rows) will be
+            removed prior to fit according to `mask_val` and `samples_train`.
+        """
+        X, y = self._mask_data(X, y, ixs=self.samples_train)
+        self.est.fit(X, y)
+        return self
+
+    def predict(self, X):
+        """Call the estimator within `self`.
+
+        This will call `self.est.predict`.
+
+        Parameters
+        ----------
+        X : array, shape (n_samples, n_features)
+            The input array for the model. Samples (rows) will be
+            removed prior to fit according to `samples_pred`.
+        """
+        X = self._mask_data(X, ixs=self.samples_pred)
```

I think we need an option here. 
* By default I would expect a prediction on all samples, not just the mask one, so as to ensure the compatibility with cross_val_score, cross_val_predict etc. 
* However I understand the usecase where you don't want trivial predictions, in which case you could have a 'predict_on' parameter set to 'all', 'not_mask' or 'zero_mask' (to predict 0s when the data is masked)